packages/editor/src/components/post-schedule/index.js

```diff
@@ -2,14 +2,59 @@
  * WordPress dependencies
  */
 import { __experimentalGetSettings } from '@wordpress/date';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { DateTimePicker } from '@wordpress/components';
-import { useRef } from '@wordpress/element';
+import { useRef, useState } from '@wordpress/element';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+function getDayOfTheMonth( date = new Date(), firstDay = true ) {
+	const d = new Date( date );
+	return new Date(
+		d.getFullYear(),
+		d.getMonth() + ( firstDay ? 0 : 1 ),
+		firstDay ? 1 : 0
+	).toISOString();
+}
+
+export default function PostSchedule() {
+	const date = useSelect(
+		( select ) => select( editorStore ).getEditedPostAttribute( 'date' ),
+		[]
+	);
+
+	const { editPost } = useDispatch( editorStore );
+	const onUpdateDate = ( postDate ) => editPost( { date: postDate } );
+
+	const [ currentMonth, setCurrentMonth ] = useState(
+		getDayOfTheMonth( date )
+	);
+
+	// Pick up published and schduled site posts.
+	const events = useSelect(
+		( select ) =>
+			(
+				select( coreStore ).getEntityRecords( 'postType', 'post', {
+					status: 'publish,future',
+					after: getDayOfTheMonth( currentMonth ),
+					before: getDayOfTheMonth( currentMonth, false ),
+					exclude: [ select( editorStore ).getCurrentPostId() ],
+				} ) || []
+			).map( ( { title, type, date: eventDate } ) => ( {
```

Comment:
Mapping inside useSelect is not a great idea as it creates a new instance on each call (so a new rerender), you should instead keep the useSelect raw (calling the selector) and have a separate useMemo to map

Reply:
addressed here https://github.com/WordPress/gutenberg/pull/29716/commits/89325df8b275fcbe36836418c85c9ade9e73767b